src/psyclone/domain/lfric/kern_stub_arg_list.py

```diff
@@ -0,0 +1,491 @@
+        specified in the kernel metadata. If supplied it also stores this
+        access in var_accesses.
+
+        :param var_accesses: optional VariablesAccessInfo instance to store \
+            the information about variable accesses.
+        :type var_accesses: \
+            :py:class:`psyclone.core.access_info.VariablesAccessInfo`
+
+        '''
+        if self._kern.mesh.properties:
+            from psyclone.dynamo0p3 import LFRicMeshProperties
+            self.extend(LFRicMeshProperties(self._kern).
+                        kern_args(stub=True, var_accesses=var_accesses))
+
+    def quad_rule(self, var_accesses=None):
+        '''Add quadrature-related information to the kernel argument list.
+        Adds the necessary arguments to the argument list, and optionally
+        adds variable access information to the var_accesses object.
+
+        :param var_accesses: optional VariablesAccessInfo instance to store \
+            the information about variable accesses.
+        :type var_accesses: \
+            :py:class:`psyclone.core.access_info.VariablesAccessInfo`
+
+        '''
+        for rule in self._kern.qr_rules.values():
+            if var_accesses is not None:
+                for var_name in rule.kernel_args:
+                    var_accesses.add_access(var_name, AccessType.READ,
+                                            self._kern)
+            self.extend(rule.kernel_args)
+
+    def indirection_dofmap(self, function_space, operator=None,
+                           var_accesses=None):
+        '''Add indirection dofmap required when applying a CMA operator. If
+        supplied it also stores this access in var_accesses.
+
+        :param function_space: the function space for which the indirect \
+            dofmap is required.
+        :type function_space: :py:class:`psyclone.domain.lfric.FunctionSpace`
+        :param operator: the CMA operator.
+        :type operator: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+        :param var_accesses: optional VariablesAccessInfo instance to store \
+            the information about variable accesses.
+        :type var_accesses: \
+            :py:class:`psyclone.core.access_info.VariablesAccessInfo`
+
+        :raises GenerationError: if no kernel argument is supplied.
+        :raises GenerationError: if the supplied kernel argument is not a \
+                                 CMA operator.
+
+        '''
+        if not operator:
+            raise GenerationError("Internal error: no CMA operator supplied.")
+        if operator.type != "gh_columnwise_operator":
+            raise GenerationError(
+                "Internal error: a CMA operator (gh_columnwise_operator) must "
+                "be supplied but got {0}".format(operator.type))
+        super(KernStubArgList, self).indirection_dofmap(function_space,
+                                                        operator,
+                                                        var_accesses)
```

Comment:
Is this actually necessary, considering that there is only one class? Afaik this is used for the reference guide? I can't build the reference guide to verify this (it aborts with "Extension error: Could not import extension autoapi.sphinx (exception: No module named sphinx)
" - that might already be at the very end of the process, since I see 'finished...' on stdout just before ... and obviously I have sphinx installed), but if I look at the code that has been produced at that time I do see the new classes in the output.

So could you clarify that this is really necessary?

Reply:
Done.